chainservice/builder.go

```diff
@@ -251,8 +251,8 @@ func (builder *Builder) buildBlockDAO(forTest bool) error {
 		dbConfig := builder.cfg.DB
 		dbConfig.DbPath = builder.cfg.Chain.ChainDBPath
 		dbConfig.CompressLegacy = builder.cfg.Chain.CompressBlock
-
-		builder.cs.blockdao = blockdao.NewBlockDAO(indexers, dbConfig)
+		blockDAOConfig, _ := blockdao.CreateBlockdaoConfig(dbConfig, blockdao.EVMNetworkIDOption(builder.cfg.Chain.EVMNetworkID))
```

Comment:
In product, `EVMNetworkID` will always be assigned to initialize `blockdao`. Therefore, it is not an option, but a parameter. Accordingly, we may define
`blockdao.NewBlockDAO(indexers, evmNetworkID, dbConfig)` and the data structs `XXXConfig` are useless.

Reply:
done